testutils/api/client.py

```diff
@@ -44,17 +51,30 @@ def call(
     ):
         url = self.__make_url(url)
         url = self.__subst_path_params(url, path_params)
-        return requests.request(
-            method,
-            url,
-            json=body,
-            data=data,
-            params=qs_params,
-            headers=self.__make_headers(headers),
-            auth=auth,
-            verify=False,
-            files=files,
-        )
+        try:
+            p = None
+            if isK8S() and url.startswith("http://mender-"):
```

Comment:
that's because when you consume public APIs, you contact https://staging.hosted.mender.io through the internet (which is wonderful, because you are actually testing the real application). When you call internal APIs, you have to port-forward using kubectl. As all our internal services start with "mender-", this if detects it and starts the kubectl port-forward only in those cases. It sleeps a few seconds because the port-forwarding is not immediate.

Reply:
Got it. Thanks!